fapolicy_analyzer/ui/fapd_manager.py

```diff
@@ -209,15 +210,16 @@ def _status(self):
             logging.debug("fapd is currently DISABLED")
             return ServiceStatus.UNKNOWN
         elif self.mode == FapdMode.ONLINE:
-            try:
-                if self._fapd_lock.acquire(blocking=False):
+            if self._fapd_lock.acquire(blocking=False):
```

Comment:
Can you use a `with` block here?

Reply:
I looked into that approach originally. The fly/ointment thing is the blocking=False argument, i.e. if the lock is not available then don't wait and return True. Fapd control operations (start/stop) are given priority over the never-ending periodic status requests.

Still digging around. I would still prefer to have locking somewhere given the status requests and stop/start requests originate on differing threads. I don't care if it's under the hood in rust-land or not.